modules/contact-form/grunion-contact-form.php

```diff
@@ -955,20 +1010,20 @@ public function personal_data_search_filter( $search ) {
 		/*
 		 * Limits search to `post_content` only, and we only match the
 		 * author's email address whenever it's on a line by itself.
-		 * `CHAR(13)` = `\r`, `CHAR(10)` = `\n`
 		 */
-		if ( preg_match( '/AUTHOR EMAIL\: ([^{\s]+)/', $search, $m ) ) {
-			$option_name = sprintf( '_jetpack_pde_feedback_%s', md5( $m[1] ) );
-			$last_post_id = get_option( $option_name, 0 );
-
-			$esc_like_email = esc_sql( $wpdb->esc_like( 'AUTHOR EMAIL: ' . $m[1] ) );
-			$search         = " AND (
-				{$wpdb->posts}.post_content LIKE CONCAT('%', CHAR(13), '{$esc_like_email}', CHAR(13), '%')
-				OR {$wpdb->posts}.post_content LIKE CONCAT('%', CHAR(10), '{$esc_like_email}', CHAR(10), '%')
-			)";
-
-			if ( $last_post_id ) {
-				$search .= $wpdb->prepare( " AND {$wpdb->posts}.ID > %d", $last_post_id );
+		if ( $this->pde_email_address && false !== strpos( $search, '..PDE..AUTHOR EMAIL:..PDE..' ) ) {
```

Comment:
We probably don't need the `strpos()` check here. Just check that it's equal to the `..PDE..AUTHOR EMAIL:..PDE..` flag.